Annotations for WebBluetooth webidl functions

zakorgy opened the pull request:
Step annotations for the WebBluetooth webidl functions, excluding `requestDevice` and unimplmeneted methods.

<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #12614

dati91 commented on components/net/bluetooth_thread.rs:
```diff
@@ -508,18 +519,26 @@ impl BluetoothManager {
             return drop(sender.send(Err(BluetoothError::Security)));
         }
+
+        // Step 2.4.
```
you lost me here :(

dati91 commented on components/net/bluetooth_thread.rs:
```diff
@@ -760,8 +815,14 @@ impl BluetoothManager {
     fn write_value(&mut self, id: String, value: Vec<u8>, sender: IpcSender<BluetoothResult<bool>>) {
         let mut adapter = get_adapter_or_return_error!(self, sender);
+
+        // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-readvalue
```
read value?

dati91 commented on components/net/bluetooth_thread.rs:
```diff
@@ -749,8 +798,14 @@ impl BluetoothManager {
         let mut adapter = get_adapter_or_return_error!(self, sender);
+
+        // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-readvalue
+        // Step 4.2.
```
There is no `4.2` in `read value`. Do you mean `5.4.2`?

dati91 commented on components/script/dom/bluetoothremotegattcharacteristic.rs:
```diff
@@ -99,14 +99,25 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
         let descriptor = receiver.recv().unwrap();
+
+        // Step 5.
+        // TODO(#4282): Transforming promise.
```
The "transforming promise" is mentioned in `Step 7`.

dati91 commented on components/script/dom/bluetoothremotegattcharacteristic.rs:
```diff
@@ -159,54 +180,101 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
             return Err(Network)
         }
+
+        // TODO(#4282): Step 4: connection-checking-wraper.
```
`Step 5`

dati91 commented on components/script/dom/bluetoothremotegattcharacteristic.rs:
```diff
@@ -159,54 +180,101 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
+
+        // TODO(#4282): Step 4: connection-checking-wraper.
+
+        // Step 4.1.
```
`Step 5.1`

shinglyu committed:
Use selenium to drive gecko test

Commit:
Move convert_request_device_options into request_bluetooth_devices.

Commit:
Update requestDevice step annotations.

shinglyu committed:
Various easy nits

shinglyu committed:
Space nit

shinglyu committed:
Remove a level of unnessary function call

shinglyu committed:
Use resource manager in gecko_driver and nits

dati91 committed:
Add event target for bluetooth

Commit:
GATTServer functions

Manishearth committed:
Add ./mach filter-intermittents for catching intermittents on CI

emilio committed:
style: Introduce StyleBloom

The idea is this will fix the bad behavior of the bloom filter in parallel
traversal.

emilio committed:
style: Enable the bloom filter recovering.

Commit:
Gatt function

Commit:
Service functions

Commit:
Characteristic functions

Commit:
Descriptor functions

nox committed:
Move to Cargo workspaces

Razican committed:
Implemented colum-rule-style

Commit:
Auto merge of #14381 - servo:workspaces, r=SimonSapin

Move to Cargo workspaces

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14381)
<!-- Reviewable:end -->

Commit:
Auto merge of #14383 - Razican:column-rule-style, r=Wafflespeanut

Stylo: implement column-rule-style

Implemented `column-rule-style` in stylo.

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #13868

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because it's a style addition and I added the screenshot.

Screenshot:
![screenshot from 2016-11-27 21-59-38](https://cloud.githubusercontent.com/assets/597469/20652731/0a7ccbfc-b4ff-11e6-8e4c-1547897bcf25.png)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14383)
<!-- Reviewable:end -->

dati91 commented on components/bluetooth/lib.rs:
```diff
@@ -754,6 +782,10 @@ impl BluetoothManager {
         };
         let services = primary_service.get_includes(device).unwrap_or(vec!());
+
+        // TODO: Step 7: If its argument is empty, throws a NotFoundError.
```
hmm how do we missed this step? we don't have a test for this or what?

zakorgy replied:
I think this step here is just an early return, if the services vec is empty, the function will return still with a `NotFoundError` [at the end of the function](https://github.com/szeged/servo/pull/57/files/c86e57239515812358fdea95dbae665769c6eb42#diff-0138552b0d86c05d27c9ded05f6f2f35R804).
In other functions e.g. in `get_primary_service` there is a check here too, I just missed that step here.

dati91 commented on components/bluetooth/lib.rs:
```diff
@@ -789,6 +823,10 @@ impl BluetoothManager {
         };
         let services = primary_service.get_includes(device).unwrap_or(vec!());
+
+        // TODO: Step 7: If its argument is empty, throws a NotFoundError.
```
Same as above

dati91 commented on components/bluetooth/lib.rs:
```diff
@@ -944,60 +1010,109 @@ impl BluetoothManager {
             Some(v) => match v {
+                // (Characteristic) Step 7.5.3.
+                // (Descriptor) Step 7.4.3.
+                // Note: There is no need to send back the value, because both algorithm returns with undefined.
```
Yes we do, because we need to set the new value [here](https://github.com/servo/servo/blob/master/components/script/dom/bluetoothremotegattcharacteristic.rs#L304).

dati91 commented on components/script/dom/bluetooth.rs:
```diff
@@ -285,47 +287,66 @@ fn canonicalize_filter(filter: &BluetoothLEScanFilterInit) -> Fallible<Bluetooth
     };
 
-    // Step 2.4.6 - 2.4.7
+    // Step 6 -7.
```
missing whitespace

dati91 commented on components/script/dom/bluetoothremotegattcharacteristic.rs:
```diff
@@ -167,18 +187,32 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
+        // TODO: Step 5: Implement the `connection-checking-wrapper` algorithm for BluetoothRemoteGATTServer.
+
+        // Step 5.1.
+        // Note: May make this check in the lower level too.
```
Why?

zakorgy replied:
Because the `properties` link in [ReadValue](https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-readvalue) Step 5.1. points [here](https://webbluetoothcg.github.io/web-bluetooth/#characteristic-properties) not to the [BluetoothCharacteristicProperties](https://webbluetoothcg.github.io/web-bluetooth/#characteristicproperties)

dati91 commented on components/script/dom/bluetoothremotegattcharacteristic.rs:
```diff
@@ -294,17 +364,42 @@ impl AsyncBluetoothListener for BluetoothRemoteGATTCharacteristic {
+                // TODO: Step 5.5.1: Implement activeAlgorithms internal slot for BluetoothRemoteGATTServer.
+
+                // Step 5.5.2.
+                // TODO(#5014): Replace ByteString with ArrayBuffer when ArrayBuffer is implemented.
```
`Replace ByteString with ArrayBuffer when it is implemented.`

dati91 commented on components/script/dom/bluetoothremotegattcharacteristic.rs:
```diff
@@ -294,17 +364,42 @@ impl AsyncBluetoothListener for BluetoothRemoteGATTCharacteristic {
+                // TODO: Step 7.5.1: Implement activeAlgorithms internal slot for BluetoothRemoteGATTServer.
+
+                // Step 7.5.2.
+                // TODO(#5014): Replace ByteString with a DataView wrapping a new ArrayBuffer
```
needs wording

Commit:
Fix a typo in bluetooth_traits/lib.rs

Commit:
Auto merge of #14392 - zakorgy:patch-2, r=Ms2ger

Fix a typo in components/bluetooth_traits/lib.rs

<!-- Please describe your changes on the following line: -->

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14392)
<!-- Reviewable:end -->

Manishearth committed:
Run filter-intermittents on buildbot

Commit:
Auto merge of #14353 - emilio:fix-bloom, r=SimonSapin

Fix the bloom filter stuff.

<!-- Please describe your changes on the following line: -->

I think I got the numbers right, want to do a try run before just in case.

r? @SimonSapin

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14353)
<!-- Reviewable:end -->

Manishearth committed:
Use jdm's tracker

Manishearth committed:
Use log- prefix for filtered log arg so that buildbot picks it up and uploads as an artefact

Commit:
Auto merge of #14331 - Manishearth:filter-intermittents, r=larsbergstrom,jdm

Add ./mach filter-intermittents for catching intermittents on CI

cc @metajack

The plan here is to run this on the error summary of the logs after each WPT/CSS run, as `./mach filter-intermittents wpt-errorsummary.log --output filtered-errorsummary.log --auth /path/to/authfile`

The `filtered-errorsummary.log` file will be a buildbot artifact for this run, and can be used for updating test results.

We change WPT/CSS runs to not cause test failures on Buildbot; instead; the test failure will be caused by this job.

We should at some point add a separate highfive task which ccs the appropriate bugs and tracks consistent failures. (We can change mach filter-intermittents to output a second file of intermittents matched to bug numbers to make this easy)

A small issue with this is that this simple task might mask failures of the WPT harness itself. We have a separate test that tests if the test harness works though. (This test will fail if the log files it requires don't exist, perhaps that's enough to solve the problem)

r? @larsbergstrom

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14331)
<!-- Reviewable:end -->

Manishearth committed:
Add a log of only intermittents

Commit:
Auto merge of #14400 - Manishearth:filter-intermittents, r=KiChjang

Add a log of only intermittents

r? @KiChjang

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14400)
<!-- Reviewable:end -->

Commit:
Auto merge of #14194 - shinglyu:gecko-webdriver, r=larsbergstrom,aneeshusa,jgraham

Use Selenium for Gecko performance test

<!-- Please describe your changes on the following line: -->
We run the `etc/ci/performance` test on Gecko for comparison, but the old add-on approach doesn't work anymore. I'm using selenium to drive Gecko instead. Also did some refactoring and fixed some nits I saw along the way.

r? @aneeshusa

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #14148  (github issue number if applicable).

<!-- Either: -->
- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14194)
<!-- Reviewable:end -->

Commit:
Auto merge of #14387 - szeged:event-target, r=jdm

Add event target for bluetooth

<!-- Please describe your changes on the following line: -->
Add support for event listeners in webbluetooth.
---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/14387)
<!-- Reviewable:end -->

Commit:
fixup! Update requestDevice step annotations.

Commit:
fixup! Descriptor functions

Commit:
fixup! Characteristic functions
